File: storage/data_transpiled/at_coder/agc016/B/3685874.py
def _import ( ) : return _import ( "" , "" )
class Main ( object ) :
    def __init__ ( self , args ) :
        super ( Main , self ).__init__ ( )
        self.args = args
    def read ( self ) :
        if not self.hasNext ( ) :
            raise StopIteration
        n = self.__next__ ( )
        if self.__next__ ( ) :
            minus = True
            self.__next__ ( )
        if self.__next__ ( ) < '0' or self.__next__ ( ) :
            raise ValueError ( )
        while self.__next__ ( ) < '0' :
            self.__next__ ( )
            self.__next__ ( )
        if self.__next__ ( ) == '0' :
            self.__next__ ( )
        if self.__next__ ( ) == - 1 or self.__next__ ( ) == - 1 or not is_printable_char ( self.__next__ ( ) ) :
            return int ( self.__next__ ( ) ) - n
        else :
            raise ValueError ( )
    def __next__ ( self ) :
        low = self.__next__ ( )
        if self.__next__ ( ) > low + 1 :
            self.__next__ ( )
            return low
        lower = 0
        for i in range ( self.__next__ ( ) ) :
            if self.__next__ ( ) == low :
                lower += 1
        higher = self.__next__ ( )
        if higher == 1 :
            self.__next__ ( )
            return low
        if higher == 0 :
            if self.lower == N - 1 or self.__next__ ( ) <= N // 2 :
                self.__next__ ( )
            else :
                self.__next__ ( )
        else :
            max = lower + ( higher // 2 )
            min = lower + 1
            if min <= low + 1 <= max :
                self.__next__ ( )
                return low
class FastScanner ( object ) :
    def __init__ ( self , buffer ) :
        self.buffer = buffer
        self.ptr = 0
        self.buflen = 0
    def __next__ ( self ) :
        if self.__next__ ( ) :
            raise StopIteration ( )
return

File: storage/data_original/at_coder/agc016/B/3685874.java
import java.io.IOException;
import java.util.Arrays;
import java.util.NoSuchElementException;

 
public class Main{

	
	public static void main(String args[])throws Exception{
		FastScanner sc = new FastScanner();
		
		int N = sc.nextInt();
		
		int[] a = new int[N];

		for(int i=0;i<N;i++){
			a[i] = sc.nextInt();
		}
		
		Arrays.sort(a);
		int low = a[0];
		if(a[N-1]>low+1){
			System.out.println("No");
			return;
		}
		
		int lower = 0;
		for(int i=0;i<N;i++){
			if(a[i]==low){
				lower++;
			}
		}
		int higher = N - lower;
		if(higher==1){
			System.out.println("No");
			return;
		}
		if(higher==0){
			if(low==N-1||low<=N/2){
				System.out.println("Yes");
			}else{
				System.out.println("No");
			}
		}else{
			
		
		int max = lower + (higher/2);
		int min = lower+1;
		if(min<=low+1&&low+1<=max){
			System.out.println("Yes");
		}else{
			System.out.println("No");
		}
		}
	}
	

	
}









class FastScanner {
    private final java.io.InputStream in = System.in;
    private final byte[] buffer = new byte[1024];
    private int ptr = 0;
    private int buflen = 0;
    private boolean hasNextByte() {
        if (ptr < buflen) {
            return true;
        }else{
            ptr = 0;
            try {
                buflen = in.read(buffer);
            } catch (IOException e) {
                e.printStackTrace();
            }
            if (buflen <= 0) {
                return false;
            }
        }
        return true;
    }
    private int readByte() { if (hasNextByte()) return buffer[ptr++]; else return -1;}
    private static boolean isPrintableChar(int c) { return 33 <= c && c <= 126;}
    private void skipUnprintable() { while(hasNextByte() && !isPrintableChar(buffer[ptr])) ptr++;}
    public boolean hasNext() { skipUnprintable(); return hasNextByte();}
    public String next() {
        if (!hasNext()) throw new NoSuchElementException();
        StringBuilder sb = new StringBuilder();
        int b = readByte();
        while(isPrintableChar(b)) {
            sb.appendCodePoint(b);
            b = readByte();
        }
        return sb.toString();
    }
    public int nextInt() {
        if (!hasNext()) throw new NoSuchElementException();
        long n = 0;
        boolean minus = false;
        int b = readByte();
        if (b == '-') {
            minus = true;
            b = readByte();
        }
        if (b < '0' || '9' < b) {
            throw new NumberFormatException();
        }
        while(true){
            if ('0' <= b && b <= '9') {
                n *= 10;
                n += b - '0';
            }else if(b == -1 || !isPrintableChar(b)){
                return (int) (minus ? -n : n);
            }else{
                throw new NumberFormatException();
            }
            b = readByte();
        }
    }
}